### ISSUE TITLE
When the folder in the dataset is mounted, data cannot be read correctly

### ISSUE THREAD

**Issue by gbdjxgp** [ultralytics/ultralytics#14672]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

My dataset is organized as follows:
```shell
dataset
|-poly
|   |-images->../images
|   |-labels
|-obb
|   |-images->../images
|   |-labels
|-images
```
the YAML file wirtes like:
```YAML
path: path/to/poly/ # dataset root dir
train: images/train # train images (relative to 'path') 4 images
val: images/val # val images (relative to 'path') 4 images
```
But when training, the program will try to find labels in the "dataset" folder instead of from the "poly" folder

### Environment

Ultralytics YOLOv8.2.62 🚀 Python-3.8.19 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 3080, 10009MiB)
Setup complete ✅ (96 CPUs, 376.3 GB RAM, 62.2/439.0 GB disk)

OS                  Linux-3.10.0-693.el7.x86_64-x86_64-with-glibc2.17
Environment         Linux
Python              3.8.19
Install             git
RAM                 376.30 GB
CPU                 Intel Xeon Platinum 8255C 2.50GHz
CUDA                12.1

numpy               ✅ 1.24.4<2.0.0,>=1.23.0
matplotlib          ✅ 3.7.5>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.10.1>=1.4.1
torch               ✅ 2.3.1>=1.8.0
torchvision         ✅ 0.18.1>=0.9.0
tqdm                ✅ 4.66.4>=4.64.0
psutil              ✅ 6.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.0.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.0>=2.0.0

### Minimal Reproducible Example

1. mount dataset like i mentioned before
2. prepare some YAML files like before
`from ultralytics import YOLO`
`results = model.train(data="YAMLfile.yaml", epochs=200, imgsz=640,val=True)`

### Additional

This method of mounting allows for easy storage of multiple formats of labels for the same image without the need to copy the image to many places. I hope this issue can be resolved.
The problematic code may be in these areas
https://github.com/ultralytics/ultralytics/blob/9df18ca9eff114743e3752ff95fc9c35d8e1202b/ultralytics/data/utils.py#L307-L316

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by ambitious-octopus** [comment]
Hey @gbdjxgp, please use the absolute path of your dataset in the `.yaml` file.

```yaml
path: /path/to/poly/ # <---- absolute path here
train: images/train # train images (relative to 'path') 4 images
val: images/val # val images (relative to 'path') 4 images
```

**Comment by gbdjxgp** [comment]
> Hey @gbdjxgp, please use the absolute path of your dataset in the `.yaml` file.
> 
> ```yaml
> path: /path/to/poly/ # <---- absolute path here
> train: images/train # train images (relative to 'path') 4 images
> val: images/val # val images (relative to 'path') 4 images
> ```

Hello, thank you for your reply. The root directory/path/to/poly of the dataset uses an absolute path (starting with/), but because the image folder (poly/images) in the absolute path is mounted to the same level directory of poly using ln - s, the labels will be read from the same level directory of poly (dataset) instead of from poly/labels, which is not intuitive.
![image](https://github.com/user-attachments/assets/7f1a8a7f-408a-49d6-9a00-06aef50d7f5d)

**Comment by Laughing-q** [comment]
@gbdjxgp Hey that's because the soft link you created pointing `datasets/poly/images` to `datasets/images` so the code would take the latter as the image folder, however we need the `labels` folder in the same root with image folder, which means the it'd assume you also have labels in `datasets/labels`(but your labels lies in `datasets/poly/labels`.

**Comment by Laughing-q** [comment]
@gbdjxgp if you want to use soft link for the data location instead of moving images or labels around the directory, I guess you could update your data.yaml based on the `dataset/` folder, something like this:
```yaml
path: datasets # dataset root dir
train: images/train # train images (relative to 'path') 4 images
val: images/val # val images (relative to 'path') 4 images
```
and in `datasets/` folder should comprise `images/` and `labels/`, and these two should be the soft link of images and labels from other folders.

**Comment by gbdjxgp** [comment]
> @gbdjxgp if you want to use soft link for the data location instead of moving images or labels around the directory, I guess you could update your data.yaml based on the `dataset/` folder, something like this:
> 
> ```yaml
> path: datasets # dataset root dir
> train: images/train # train images (relative to 'path') 4 images
> val: images/val # val images (relative to 'path') 4 images
> ```
> 
> and in `datasets/` folder should comprise `images/` and `labels/`, and these two could be the soft links pointing other locations of the images and labels you want to train.
> 
> EDIT: I just tested and it works to me!
@Laughing-q Thank you very much for your detailed reply and answer!
I just prepare the dataset like this, both the images and labels are mounted into the test_dataset folder and the test_dataset is not a soft link folder, both images and labels are soft link folder linked to other locations. It didn's works for me.
![image](https://github.com/user-attachments/assets/0bf6ae5e-7b22-479e-9966-22ef20cf3547)
Since i just want to storage a single image with multiple labels, maybe the case you mentioned is not suit for me,(Because I always try to mount the labels folder to a directory at a different level than images). I made changes to the code as shown in the picture below, which worked for me.
![image](https://github.com/user-attachments/assets/c0739fdb-b4a1-4d24-8dea-24f1f91cea32)
